### ISSUE TITLE
False positives in zero_prefixed_literal and `0b`/`0o`

### ISSUE THREAD

**Issue by dtolnay** [Manishearth/rust-clippy#1202]
```rust
fn main() {
    let _ = 0b1111;
    let _ = 0o7777;
}
```

```
warning: this is a decimal constant, #[warn(zero_prefixed_literal)] on by default
 --> src/main.rs:2:13
  |
2 |     let _ = 0b1111;
  |             ^^^^^^
  |
help: if you mean to use a decimal constant, remove the `0` to remove confusion:
  |     let _ = b1111;
help: if you mean to use an octal constant, use `0o`:
  |     let _ = 0ob1111;
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#zero_prefixed_literal

warning: this is a decimal constant, #[warn(zero_prefixed_literal)] on by default
 --> src/main.rs:3:13
  |
3 |     let _ = 0o7777;
  |             ^^^^^^
  |
help: if you mean to use a decimal constant, remove the `0` to remove confusion:
  |     let _ = o7777;
help: if you mean to use an octal constant, use `0o`:
  |     let _ = 0oo7777;
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#zero_prefixed_literal
```

@mcarton

**Comment by mcarton** [comment]
Oh, that's embarrassing. On it.

**Comment by mcarton** [comment]
Thanks for that fast report :smile:
